circuits/utils/circuitFunctionGetter.ts

```diff
@@ -0,0 +1,17 @@
+import {
+  CircuitFunctions,
+  circuitFunctionsArray,
+} from "../types/circuitFunctions.types";
+
+export function getCircuitFunctionIndex(funcName: CircuitFunctions) {
+  const index = circuitFunctionsArray.indexOf(funcName);
+  if (index === -1)
+    throw new Error(`function with name ${funcName} does not exist`);
+  return index;
+}
```

Comment:
In theory this is worse than an enum because `array.indexOf` is O(n) complexity 🤣 the number enums in TS will give you a reverse mapping which allows for O(1) lookup from index and from string. I'm not fussed about this tho - just an observation

Reply:
lol 😂